### PR TITLE
Implemenation of the Serialized Lob Pattern

### PR DESCRIPTION
**Pull request by KishalayP**
Pull request title

- #1596 : Implemenation of the Serialized LOB Pattern as described in the issue

Pull request description

- Added the module slob
- Added both serializers mentioned in EAA, the Clob Serializer and Blob Serializer.
- Implmented DB Service to persist objects to H2 DB.
- Added required Java Docs and updated readme.
- Added Unit Tests for both the serializers

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Comment by KishalayP** [comment]
@iluwatar  completed the implmentation, could you please review the PR.

**Comment by KishalayP** [comment]
Hi @iluwatar completed the implemntation for slob, can you please review the PR.

File: slob/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Serialized LOB
+category: Structural
+language: en
+tag:
+  - Object Instantiation
```
Comment (iluwatar) [inline slob/README.md:6]:
```suggestion
  - Data access
  - Instantiation
```

File: slob/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+* The Forest here represents the object graph.
+* A forest contains animals and plants. As is in real life the forest object contains plants and
+  animals
+  where some animals eat other animals and some eat only plants.
+* These relationships are maintained in the Forest Object.
+* There are 2 types of Serializers available ClobSerializer and BlobSerializer.
+* ClobSerializer uses character or textual based serialization, here the Object graph is converted
+  to XML and then stored as text in DB.
+* BlobSerializer uses binary data for serialization, here the Object graph is converted to Byte
+  Array and then stored as a blob in DB.
```
Comment (iluwatar) [inline slob/README.md:28]: Please follow the pattern template: https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template

See the other patterns for examples

File: slob/src/main/java/com/iluwatar/slob/App.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import java.util.Set;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+/**
+ * SLOB Application using {@link LobSerializer} and H2 DB.
+ */
```
Comment (iluwatar) [inline slob/src/main/java/com/iluwatar/slob/App.java:46]: Here above the App class, please describe the pattern briefly and explain how the example implements it

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: slob/README.md (hunk 1 above)
Comment (KishalayP) [reply slob/README.md]: Have updated, please review

File: slob/README.md (hunk 2 above)
Comment (KishalayP) [reply slob/README.md]: Updated as per the template, please review

File: slob/src/main/java/com/iluwatar/slob/App.java (hunk 3 above)
Comment (KishalayP) [reply slob/src/main/java/com/iluwatar/slob/App.java:46]: Described the pattern briefly here and explained how the implmentation works, please review

File: slob/README.md [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+import java.util.HashSet;
+import java.util.Set;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
```
Comment (iluwatar) [inline slob/README.md:49]: The code snippets included in the `README.md` should be minimal. We don't want to include irrelevant parts of the code, just the things needed to explain the pattern. See the other patterns for examples of how to do this.

File: slob/src/main/java/com/iluwatar/slob/App.java (hunk 3 above)
Comment (iluwatar) [reply slob/src/main/java/com/iluwatar/slob/App.java:46]: Maybe its still a little bit too brief. From the reader's point of view

- What is the SLOB design pattern? (intent of the pattern)
- How is it implemented in this code example? (structure, class responsibilities, etc.)

For an example, see here: https://github.com/iluwatar/java-design-patterns/blob/master/abstract-factory/src/main/java/com/iluwatar/abstractfactory/App.java#L30-L43

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: slob/README.md (hunk 4 above)
Comment (KishalayP) [reply slob/README.md]: Got it, removed import statmenets and license doc from README.md. Kindly check now

**Review by iluwatar — CHANGES_REQUESTED** [review]
Noticed that the pattern has not been added to the parent pom.xml, so it is not built by the CI.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @KishalayP for code

**Comment by KishalayP** [comment]
Thank You @iluwatar .

> Looks good! Thank you for the contribution 🎉
> 
> @all-contributors please add @KishalayP for code

**Pull request merged**